src/vs/workbench/api/browser/mainThreadDocuments.ts

```diff
@@ -30,14 +31,30 @@ export class BoundModelReferenceCollection {
 		//
 	}
 
-	dispose(): void {
+	dispose(uri?: URI): void {
+		if (uri) {
```

Comment:
This isn't really needed, right? I am more in favour of keeping `dispose` as-is and only have the special dispose-that-one-resource function.

Reply:
@jrieken yeah I pushed https://github.com/microsoft/vscode/pull/98154/commits/cd68bcb0272c0a681fb75a660d7e7798ec247ce9 to have an explicit `remove` method.

I wonder if the use of `isEqualOrParent` is a little bit fishy in this context because it should only be used for URIs that have file system semantics, not for any other. Let me know how you feel about it, we could make that conditional only if the scheme is known to the file service.

Reply:
Yeah, it took me a while to understand what this is about at all. I'd say it unlikely to cause false positives